test/ibmq/test_ibmq_jobmanager.py

```diff
@@ -283,6 +287,65 @@ def test_skipped_result(self, provider):
         else:
             self.log.warning("Unable to cancel job %s", cjob.job_id())
 
+    def test_ibmq_managed_results_signature(self):
+        """Test `ManagedResults` and `Result` contain the same public methods.
+
+        Note:
+            Aside from ensuring the two classes contain the same public
+            methods, it is also necessary to check that the corresponding
+            methods have the same signature.
+        """
+        # Get `Result` public methods.
+        result_methods = self._get_class_methods(Result)
+        self.assertTrue(result_methods)
+
+        # Get `ManagedResults` public methods.
+        managed_results_methods = self._get_class_methods(ManagedResults)
+        self.assertTrue(managed_results_methods)
+
+        # Ensure `ManagedResults` does not have any additional public methods.
+        additional_methods = set(managed_results_methods.keys()) - set(result_methods.keys())
+        self.assertTrue((len(additional_methods) == 0),
+                        "ManagedResults does not have the same public "
+                        "methods as Result. ManagedResults has the "
+                        "additional public method(s): {}"
+                        .format(additional_methods))
+
+        # Ensure `ManagedResults` is not missing any public methods.
+        missing_methods = set(result_methods.keys()) - set(managed_results_methods.keys())
+        self.assertTrue((len(missing_methods) == 0),
```

Comment:
I think you can combine this check and the one above with a single `assertEqual`, which _may_ even print out the differences if the assert fails.